subprojects/core/src/main/java/org/gradle/api/internal/classpath/DefaultModuleRegistry.java

```diff
@@ -97,6 +98,9 @@ public Module getExternalModule(String name) {
     private Module loadExternalModule(String name) {
         File externalJar = findJar(name);
         if (externalJar == null) {
+            if (gradleInstallation == null) {
+                throw new UnknownModuleException(String.format("Cannot locate JAR for module '%s'. Gradle Home isn't present", name));
+            }
```

Comment:
Yes, this was null in one of my tests when I didn't have `executer.requireGradleDistribution()`.

Reply:
Related: https://github.com/gradle/gradle/issues/7413